pennylane/operation.py

```diff
@@ -476,11 +488,34 @@ def generator(self):
         """
         return [None, 1]
 
+    @property
+    def is_inverse(self):
+        """Boolean determining if the inverse of the operation was requested.
+        """
+        return str.endswith(self._name, self.string_for_inverse)
```

Comment:
I think this solution is not elegant because you save the information if the operation was inverted or not (which is boolean by nature) into the name. 

I would recommend to use the boolean property `_inverse` as a toggle that is flipped when `.inv()` is called. In the property `name` you can then have something along the lines of 
```python
return self._name + ".inv" if self._inverse else self._name
```

Reply:
If you decide to do this you would have to be careful of where to use `.name` vs `._name` in the rest of the codebase.

Reply:
I also prefer this design, incorporated your suggestions. Thanks!